src/foremast/s3/s3deploy.py

```diff
@@ -88,20 +89,26 @@ def _path_formatter(self, suffix):
     def upload_artifacts(self):
         """Upload artifacts to S3 and copy to LATEST depending on strategy."""
         deploy_strategy = self.properties[self.env]["deploy_strategy"]
+        self._upload_artifacts_to_version()
         if deploy_strategy == "highlander":
-            self._upload_artifacts_to_version()
             self._sync_to_uri(self.s3_latest_uri)
-        elif deploy_strategy == "redblack":
-            self._upload_artifacts_to_version()
         elif deploy_strategy == "canary":
-            self._upload_artifacts_to_version()
             self._sync_to_uri(self.s3_canary_uri)
+        elif deploy_strategy == "alpha":
+            self._sync_to_uri(self.s3_alpha_uri)
         else:
             raise NotImplementedError
 
-    def promote_artifacts(self):
-        """Promote artifact version to LATEST."""
-        self._sync_to_uri(self.s3_latest_uri)
+    def promote_artifacts(self, promote_stage='latest'):
+        """Promote artifact version to dest.
+
+        Args:
+            promote_stage (string): Stage that is being promoted
+        """
+        if promote_stage.lower() == 'alpha':
+            self._sync_to_uri(self.s3_canary_uri)
```

Comment:
Could we make it a bit explicit by also adding 'canary' and what happens? Its safe to assume today that if not 'alpha' it should go to LATEST today, but long term I can see this becoming a bug if we or anyone adds more stages.

Reply:
Thanks!